### DISCUSSION TITLE
Message getting lost without error or warning

### DISCUSSION THREAD

**Discussion by shkelzeen** [rabbitmq/rabbitmq-delayed-message-exchange#179] [General]
I have a scenario as follow. I have an exchange `E` which is bind with queue `Q`, have a scheduler which puts around 1000 messages on the exchange and all the messages are passed to the queue and I receive them on the listener, while I am processing the messages if a error happens messages are send to a other fanout  delayed exchange `FD` with a delay of 60s and this process is repeated until all the messages are processed. The issue is for some reason some messages which are being put on `FD` are getting lost with error warning

**Comment by michaelklishin** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(

**Comment by michaelklishin** [comment]
We don't have enough information to suggest anything specific. There are no details on what "the error" is. You can inspect how many messages are scheduled for delivery in a given exchange. The exchange does not replicate any of the data it stores.
All routing and [publisher safety](https://www.rabbitmq.com/publishers.html#data-safety) topics are particularly relevant to this exchange since it cannot know if there will be any queues bound to it or connections to return unroutable messages to if routing fails.
Without an executable way to reproduce, we can only guess and we do not guess in this community.

Both the number of "scheduled" messages and exchange publishing rates will help narrow things down. 

This plugin has [very important limitations](https://github.com/rabbitmq/rabbitmq-delayed-message-exchange#limitations)

**Comment by shkelzeen** [comment]
I tried to create a diagram which shows what and how we are working on this

![Untitled Diagram drawio (1)](https://user-images.githubusercontent.com/12429342/141471453-064ef17a-6bc9-4d11-8da6-f46173b0ccd4.png)
